src/Framework/TaskStartedEventArgs.cs

```diff
@@ -135,6 +186,11 @@ internal override void CreateFromStream(BinaryReader reader, int version)
         /// </summary>
         public int ColumnNumber { get; internal set; }
 
+        /// <summary>
+        /// Full name of the assembly that implements the task
+        /// </summary>
+        public AssemblyName TaskAssemblyName { get; private set; }
```

Comment:
Please consider representing the full name with string here. `AssemblyName` can be [expensive to handle](https://github.com/dotnet/msbuild/blob/8fa24d94b5a0a0b164e04ea3eeb8fd102584ee5b/src/Shared/AssemblyNameExtension.cs#L50C1-L54C19) and the fewer conversions to/from string we do, the better. Also, prior art uses strings: https://github.com/dotnet/msbuild/blob/8fa24d94b5a0a0b164e04ea3eeb8fd102584ee5b/src/Framework/AssemblyLoadBuildEventArgs.cs#L39

Reply:
No problem. I checked the implementation and it's mutable, so the `FullPath` is constructed every time without caching :| I didn't expect that.